cartographer_ros/src/cartographer_node_main.cc

```diff
@@ -304,8 +304,16 @@ void Node::AddOdometry(int64 timestamp, const string& frame_id,
         Eigen::Matrix3d::Zero(), rotational;
     // clang-format on
   }
-  map_builder_.GetTrajectoryBuilder(kTrajectoryBuilderId)
-      ->AddOdometerPose(time, pose, applied_covariance);
+  try {
+    const Rigid3d sensor_to_tracking =
+        LookupToTrackingTransformOrThrow(time, odometry.child_frame_id);
+    map_builder_.GetTrajectoryBuilder(kTrajectoryBuilderId)
+        ->AddOdometerPose(time, sensor_to_tracking * odometry.pose,
```

Comment:
Since odometry.pose is in frame_id according to the documentation, I'd think that the pose is the position of child_frame_id in it, should this be: odometry.pose * sensor_to_tracking.inverse()?